### PR TITLE
Feat/delete sdk

### PR DESCRIPTION
**Pull request by zenovore**
**What this PR does / why we need it**: This PR address the SDK changes from #389, contain api and sdk client regenerated + update on model version and model deletion on the SDK

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
1. Release delete model and model version on the SDK
```

**Checklist**

- [x] Added unit test, integration, and/or e2e tests
- [x] Tested locally
- [ ] Updated documentation
- [x] Update Swagger spec if the PR introduce API changes
- [x] Regenerated Golang and Python client if the PR introduce API changes

### CODE REVIEW

File: python/sdk/merlin/model.py [hunk 1]
```diff
@@ -619,6 +619,17 @@ def set_traffic(self, traffic_rule: Dict['ModelVersion', int]) \
 
         return ModelEndpoint(ep)
 
+    def delete_model(self) -> int:
+        """
+        Delete this model. Please note that any inactive related entity (model versions, endpoints and prediction jobs) will get deleted by this process.
+        Deletion will failed if there are any active related entity (model versions, endpoints and prediction jobs)
```
Comment (krithika369) [inline python/sdk/merlin/model.py:625]: Nit: Here and in other instances, `Deletion will failed` -> `Deletion will fail`

**Review by krithika369 — APPROVED** [review]
LGTM. Please re-base again main after the API PR https://github.com/caraml-dev/merlin/pull/389 is merged, before merging this one. Thanks, @zenovore!

**Comment by deadlycoconuts** [comment]
Hey @zenovore, could you also add some description to the PR? 😅 It's still the raw template for now 👀

**Review by deadlycoconuts — APPROVED** [review]
LGTM! Thanks for the changes! 🚀

**Pull request merged**